### DISCUSSION TITLE
Make naming convention consistent for testmod directories?

### DISCUSSION THREAD

**Discussion by ekluzek** [ESCOMP/CTSM#2038] [Ideas]
@johnpaulalex noticed we aren't consistent in our naming convention for our testmod directories. We are probably closest to Pascal-Case, but we also use underscores sometimes, and leave acronyms in uppercase. We talked about this as a group some. Several of us agree that consistency is helpful. Sometimes it feels like adding an underscore can help readability. It also feels like this isn't something critical to fix. So we are all over the map on this, and we'd need to come to a consensus.

**Comment by ekluzek** [comment]
@adrifoster also notes that we aren't consistent in other spaces either (python code, XML, or FORTRAN code). Those would all be separate discussions as well and more work to get consistent than here. It's worth talking about though. And I suggest that we include that in this discussion as well if people have thoughts on this.

The other note that @adrifoster makes is that because FORTRAN is case insensitive, any conventions we make need to take that into account. Changing the case is still helpful for readability in FORTRAN code, but since the compiler doesn't recognize it, it becomes problematic...

**Reply by samsrabin** [reply]
I think I've come around to PascalCase being a good choice, with the FATES test names Adrianna pointed to being nice to look at. But it does bring some limitations as you noted Erik; sometimes it's nice to be able to add an underscore. And of course we can't use hyphens in test names because CIME interprets that as a path separator.

The point about capitalization being potentially problematic in FORTRAN is a good one, although I'm not sure whether that applies to the testing framework specifically.
